src/XPurchase/VirtoCommerce.XPurchase/Schemas/GiftItemType.cs

```diff
@@ -0,0 +1,74 @@
+using System.Linq;
+using GraphQL.DataLoader;
+using GraphQL.Resolvers;
+using GraphQL.Types;
+using MediatR;
+using VirtoCommerce.ExperienceApiModule.Core.Extensions;
+using VirtoCommerce.ExperienceApiModule.Core.Helpers;
+using VirtoCommerce.ExperienceApiModule.Core.Schemas;
+using VirtoCommerce.ExperienceApiModule.Core.Services;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.XDigitalCatalog;
+using VirtoCommerce.XDigitalCatalog.Queries;
+using VirtoCommerce.XDigitalCatalog.Schemas;
+
+namespace VirtoCommerce.XPurchase.Schemas
+{
+    public class GiftItemType : ExtendableGraphType<GiftItem>
+    {
+        public GiftItemType(IMediator mediator, IDataLoaderContextAccessor dataLoader, IDynamicPropertyResolverService dynamicPropertyResolverService)
+        {
+            Field(x => x.PromotionId).Description("Promotion Id");
+            Field(x => x.Quantity).Description("Quantity of gifts in the reward");
+            Field(x => x.ProductId, true).Description("Product id");
+            Field(x => x.CategoryId, true).Description("Product Category Id");
+            Field(x => x.ImageUrl, true).Description("Value of reward image absolute URL");
+            Field(x => x.Name).Description("Name of the reward");
+            Field(x => x.MeasureUnit, true).Description("Measure Unit");
+            Field(x => x.LineItemId, true).Description("ID of lineItem if gift is in cart. Otherwise null");
+
+            AddField(new FieldType
+            {
+                Name = "id",
+                Description = "Artificial ID for this value object",
+                Type = GraphTypeExtenstionHelper.GetActualType<NonNullGraphType<StringGraphType>>(),
+                Resolver = new FuncFieldResolver<string>(context =>
+                {
+                    // return value object hash as Id
+                    return context.Source.GetHashCode().ToString();
```

Comment:
You can't use hash as Id since it can be different on multiple machines. Why to don not use pair PromotionId + ProductId ?? Name

Reply:
As GiftItem is ValueObject, it's HashCode is determined by the values returned form GetEqualityComponents(). Always the same value for same Equality Components.

Reply:
HashCode newer can be used for equality check for object, the different string combination can produce the same hash code because of hashcode is number!

Reply:
makes sense, will use CacheKey then